src/SFA.DAS.EmployerApprenticeshipsService.Infrastructure/Data/TransactionRepository.cs

```diff
@@ -139,6 +140,35 @@ public TransactionRepository(LevyDeclarationProviderConfiguration configuration,
             return result.ToList();
         }
 
+        public async Task<List<TransactionDownloadLine>> GetAllTransactionDetailsForAccountByDate(long accountId, DateTime fromDate, DateTime toDate)
+        {
+            var result = await WithConnection(async c =>
+            {
+                var parameters = new DynamicParameters();
+                parameters.Add("@AccountId", accountId, DbType.Int64);
+                parameters.Add("@fromDate", fromDate, DbType.DateTime);
+                parameters.Add("@toDate", toDate, DbType.DateTime);
+
+                return await c.QueryAsync<TransactionDownloadLine>(
+                    sql: "[employer_financial].[GetAllTransactionDetailsForAccountByDate]",
+                    param: parameters,
+                    commandType: CommandType.StoredProcedure);
+            });
+
+            var hmrcDateService = new HmrcDateService();
+            var list = result.ToList();
+            foreach (var res in list)
```

Comment:
I think with the previous `.ToList()` that it's enumerating the results twice here.

Reply:
Actioned